spacer/task_utils.py

```diff
@@ -109,7 +109,7 @@ def preprocess_labels(
         ref_done = False
 
         for image_index, image_key in enumerate(labels_in.image_keys):
-            this_image_labels = labels_in.data[image_key]
+            this_image_labels = labels_in[image_key]
```

Comment:
Here an elsewhere, doing `[image_key]` rather than `.get(image_key)` means a KeyError exception if for some reason the key doesn't exist -- that's probably what you want, but just checking; you could handle or assign a default if that would help debugging.

Reply:
Good to check - I think in each of these cases it's getting `image_key` by iterating over the `image_keys` property of the same `ImageLabels` object, so it ought to be safe enough IMO; I also think there isn't really a reasonable default value to use, so it may as well raise a KeyError if the key's not there.